### PR TITLE
Unfinished example of a RADS to ioda converter

### LINKED ISSUE #607: Create template and example based on RADS altimetry to ioda

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
This PR adds a prototype IODA converter that will be used as a template for the other netdcf observation files that will need to be converted. 
I added a base class for the conversion in ```NetCDFToIodaConverter.h```, it probably could do more but for now the only thing useful it does is saving the ioda file.

### RADS to IODA implementation
- The derived class from the above that actually does something useful is called Rads2Ioda and it implements the reading of the RADS netcdf files, which is a virtual function of ```NetCDFToIodaConverter```
- The application code is under obsproc/applications

### Testing
- I added a sub-sampled cdl RADS file that is converted back to netcdf in a ctest
- The ctest reads the RADS netcdf file and creates a IODA file. It's not testing anything more than successful completion
- The converter is not complete, it's missing most of the MetaData and a few other things ... 

### Issues addressed
- fixes #607

### CODE REVIEW

File: utils/obsproc/NetCDFToIodaConverter.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#ifndef GDAS_UTILS_OBSPROC_NETCDFTOIODACONVERTER_H_  //NOLINT
```
Comment (CoryMartin-NOAA) [inline utils/obsproc/NetCDFToIodaConverter.h]:
```suggestion
#pragma once
```

File: utils/obsproc/Rads2Ioda.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#include "NetCDFToIodaConverter.h"
+
+#include "oops/util/missingValues.h"
+
+namespace gdasapp {
+  class Rads2Ioda : public NetCDFToIodaConverter {
+   public:
+    explicit Rads2Ioda(const eckit::Configuration & fullConfig)
+    : NetCDFToIodaConverter(fullConfig) {
+    oops::Log::info() << this->windowBegin_ << std::endl;
```
Comment (CoryMartin-NOAA) [inline utils/obsproc/Rads2Ioda.h]: How about some strings that say 'window begin:', etc.

File: utils/obsproc/NetCDFToIodaConverter.h (hunk 1 above)
Comment (CoryMartin-NOAA) [reply utils/obsproc/NetCDFToIodaConverter.h]: Just pragma once and remove all the ifndef stuff

File: utils/obsproc/Rads2Ioda.h (hunk 2 above)
Comment (guillaumevernieres) [reply utils/obsproc/Rads2Ioda.h]: ho ... in the log info @CoryMartin-NOAA ? Yes, good point.

File: utils/obsproc/NetCDFToIodaConverter.h (hunk 1 above)
Comment (guillaumevernieres) [reply utils/obsproc/NetCDFToIodaConverter.h]: Ha! yes ... dummy me. Are we sure it's doing the same as the ifndef stuff?

Comment (CoryMartin-NOAA) [reply utils/obsproc/NetCDFToIodaConverter.h]: Indeed, ask ChatGPT to confirm though :-)

Comment (guillaumevernieres) [reply utils/obsproc/NetCDFToIodaConverter.h]: Just did, the only issue is portability. It's not part of the c++ standard but it's widely supported. Works for me!

**Review by CoryMartin-NOAA — APPROVED** [review]
image

**Comment by guillaumevernieres** [comment]
I think the test will fail, there's a bug in the oops develop that was probably cloned by the ci ... The culprit pr was just reverted ~30mn ago. Zut!

**Review by ShastriPaturi — APPROVED** [review]

**Review by apchoiCMD — APPROVED** [review]

**Review by AndrewEichmann-NOAA — APPROVED** [review]
Looks ok - what's a cdl file?

**Comment by CoryMartin-NOAA** [comment]
@AndrewEichmann-NOAA it's a text file in the same format as what ncdump dumps out that you can use to create a netCDF file.

**Review by apchoiCMD — APPROVED** [review]

**Pull request merged**